src/pages/iou/MoneyRequestModal.js

```diff
@@ -0,0 +1,499 @@
+                props.currentUserPersonalDetails.login,
+                amount,
+                trimmedComment,
+                props.iou.selectedCurrencyCode,
+                props.preferredLocale,
+                reportID,
+            );
+            return;
+        }
+
+        // If the IOU is created from the global create menu, we also navigate the user to the group report
+        if (props.hasMultipleParticipants) {
+            IOU.splitBillAndOpenReport(
+                selectedParticipants,
+                props.currentUserPersonalDetails.login,
+                amount,
+                trimmedComment,
+                props.iou.selectedCurrencyCode,
+                props.preferredLocale,
+            );
+            return;
+        }
+        IOU.requestMoney(
+            props.report,
+            Math.round(amount * 100),
+            props.iou.selectedCurrencyCode,
+            props.currentUserPersonalDetails.login,
+            selectedParticipants[0],
+            trimmedComment,
+        );
+    }
+
+    function renderHeader() {
+        return (
+            <View style={[styles.headerBar]}>
+                <View style={[
+                    styles.dFlex,
+                    styles.flexRow,
+                    styles.alignItemsCenter,
+                    styles.flexGrow1,
+                    styles.justifyContentBetween,
+                    styles.overflowHidden,
+                ]}
+                >
+                    {currentStepIndex > 0
+                        && (
+                        <View style={[styles.mr2]}>
+                            <Tooltip text={props.translate('common.back')}>
+                                <TouchableOpacity
+                                    onPress={() => navigateToPreviousStep}
+                                    style={[styles.touchableButtonImage]}
+                                >
+                                    <Icon src={Expensicons.BackArrow} />
+                                </TouchableOpacity>
+                            </Tooltip>
+                        </View>
+                        )}
+                    <Header title={getTitleForStep()} />
+                    <View style={[styles.reportOptions, styles.flexRow, styles.pr5]}>
+                        <Tooltip text={props.translate('common.close')}>
+                            <TouchableOpacity
+                                onPress={() => Navigation.dismissModal()}
+                                style={[styles.touchableButtonImage]}
+                                accessibilityRole="button"
+                                accessibilityLabel={props.translate('common.close')}
+                            >
+                                <Icon src={Expensicons.Close} />
+                            </TouchableOpacity>
+                        </Tooltip>
+                    </View>
+                </View>
+            </View>
+        );
+    }
```

Comment:
Pretty much all of these functions should use `useCallback()` if they have dependencies.

Reply:
And if they don't we can move them to the top of the file. I looked briefly and I think most do have dependencies.

Reply:
What's the guideline for when to cache methods? I didn't think it was necessary here because none of these functions perform any heavy computational tasks that need to be cached